### ISSUE TITLE
Custom Links are not drawn on paper after redo or graph.fromJSON()

### ISSUE THREAD

**Issue by EduardHuck** [clientIO/joint#1176]
Hello,

I have a problem with my custom link regarding the recreation of it through joint/rappid functions.

Creation through the user/GUI works just fine.

But if I use `commandManager.redo()` or `graph.fromJSON(treeGraphJSON`) the links are visually missing.
They can still be found in the HTML-DOM but are all positioned at 0x0 and have the size of 0x0.

Heres my custom Link Class:

```
export class AnswerShapeBase extends joint.shapes.standard.Link {
  defaults(): Backbone.ObjectHash {
    return joint.util.defaultsDeep(
      {
        type: 'customShapes.AnswerShape',
        markup: [...],
        attrs: {...},
        defaultLabel: {...},
        labels: [...]
      },
      joint.shapes.standard.Link.prototype.defaults
    );
  }
```

After I draw a custom link on my paper and run the following code:

```
    console.log('Before', this.graph.toJSON());
    this.graph.fromJSON(this.graph.toJSON());
    console.log('After', this.graph.toJSON());
```

I get the following output regarding the custom links:

**Before:**

```
3:
attrs: {line: {…}, wrapper: {…}}
defaultLabel: {markup: Array(2), attrs: {…}}
id: "e77c3671-5122-495a-868a-fdbbbd9b7044"
labels: [{…}]
markup: (2) [{…}, {…}]
source: {id: "2685ce18-dc7b-42fe-83cd-b07987a70214"}
target: {id: "a8228829-1108-4f5c-b5e3-22bd9802d789"}
type: "customShapes.AnswerShape"
z: 4
__proto__: Object
length: 4
__proto__: Array(0)
```

**After:**

```
3:
angle: 0
attrs: {line: {…}, wrapper: {…}}
defaultLabel: {markup: Array(2), attrs: {…}}
id: "e77c3671-5122-495a-868a-fdbbbd9b7044"
labels: [{…}]
markup: (2) [{…}, {…}]
position: {x: 0, y: 0}
size: {width: 1, height: 1}
source: {id: "2685ce18-dc7b-42fe-83cd-b07987a70214"}
target: {id: "a8228829-1108-4f5c-b5e3-22bd9802d789"}
type: "customShapes.AnswerShape"
z: 4
__proto__: Object
length: 4
__proto__: Array(0)
```

The fromJSON() method adds new properties `position` and `size`, which might cause the issue.

I already tried extending joint.dia.Link instead of joint.shapes.standard.Link, which had the same result.

Thanks in advance.

**Comment by kumilingus** [comment]
I am not sure if this ES6 Class syntax works well with Backbone. Have you tried `joint.shapes.standard.Link.extend({})`?

**Comment by EduardHuck** [comment]
@kumilingus : Thanks for your fast reply. After a further analysis of the problem, I found the cause of the problem. Fortunately the problem wasn't in the typescript class extend functionality.

I was missing the `cellNamespace` and `cellViewNamespace` options on the graph and paper. 

Because of that everytime I used redo or fromJSON, the type information of my Elements and Links was lost and they were of unknown 'child' type.

The fixed code looks like this now:

**Custom-Shapes Namespace:**

```
export namespace customShapes {
  export class XYZShape extends XYZShapeBase {}
  ...
  ...
}
```

**Graph:** (https://resources.jointjs.com/docs/jointjs/v3.0/joint.html#dia.Graph.constructor)
```
import * as customShapesNamespace from '../shapes/custom-shapes';
new joint.dia.Graph({}, { cellNamespace: customShapesNamespace });
```


**Paper:** (https://resources.jointjs.com/docs/jointjs/v3.0/joint.html#dia.Paper.prototype.options.cellViewNamespace)

```
import * as customShapesNamespace from '../shapes/custom-shapes';

new joint.dia.Paper({ ... ,
    cellViewNamespace: customShapesNamespace
  });
```
